jobs/transforms/skill_validation_test.py

```diff
@@ -0,0 +1,290 @@
+            base_validation_errors.CommitCmdsValidateError(
+                invalid_commit_cmd_model,
+                {'invalid': 'data'},
+                'Missing cmd key in change dict')
+        ])
+
+    def test_skill_change_object_with_invalid_cmd(self):
+        invalid_commit_cmd_model = skill_models.SkillCommitLogEntryModel(
+            id='123',
+            created_on=self.YEAR_AGO,
+            last_updated=self.NOW,
+            commit_type='test-type',
+            user_id='',
+            skill_id='123',
+            post_commit_status='private',
+            commit_cmds=[{'cmd': 'invalid'}])
+
+        output = (
+            self.pipeline
+            | beam.Create([invalid_commit_cmd_model])
+            | beam.ParDo(
+                skill_validation.ValidateSkillCommitCmdsSchema())
+        )
+
+        self.assert_pcoll_equal(output, [
+            base_validation_errors.CommitCmdsValidateError(
+                invalid_commit_cmd_model,
+                {'cmd': 'invalid'},
+                'Command invalid is not allowed')
+        ])
+
+    def test_skill_change_object_with_missing_attribute_in_cmd(self):
+        invalid_commit_cmd_model = skill_models.SkillCommitLogEntryModel(
+            id='123',
+            created_on=self.YEAR_AGO,
+            last_updated=self.NOW,
+            commit_type='test-type',
+            user_id='',
+            skill_id='123',
+            post_commit_status='private',
+            commit_cmds=[{
+                'cmd': 'update_skill_property',
+                'property_name': 'name',
+            }])
+
+        output = (
+            self.pipeline
+            | beam.Create([invalid_commit_cmd_model])
+            | beam.ParDo(
+                skill_validation.ValidateSkillCommitCmdsSchema())
+        )
+
+        self.assert_pcoll_equal(output, [
+            base_validation_errors.CommitCmdsValidateError(
+                invalid_commit_cmd_model,
+                {
+                    'cmd': 'update_skill_property',
+                    'property_name': 'name',
+                },
+                'The following required attributes are missing: '
+                'new_value, old_value')
+        ])
+
+    def test_skill_change_object_with_extra_attribute_in_cmd(self):
+        invalid_commit_cmd_model = skill_models.SkillCommitLogEntryModel(
+            id='123',
+            created_on=self.YEAR_AGO,
+            last_updated=self.NOW,
+            commit_type='test-type',
+            user_id='',
+            skill_id='123',
+            post_commit_status='private',
+            commit_cmds=[{
+                'cmd': 'add_skill_misconception',
+                'new_misconception_dict': '{u\'id\': 0, u\'notes\': '
+                                          'u\'<p>notes</p>\', u\'feedback\': '
+                                          'u\'<p>default_feedback</p>\', '
+                                          'u\'name\': u\'name\'}',
```

Comment:
Please add a comment explaining why is this a string.

Reply:
The same in other places.

Reply:
Done PTAL! the other two that I have mentioned is fixed by simply rearranging the dict.